exonum-java-binding-core/rust/integration_tests/tests/transaction_proxy.rs

```diff
@@ -162,50 +153,60 @@ fn execute_should_return_err_if_tx_exec_exception_occurred_no_message() {
 #[test]
 fn execute_should_return_err_if_tx_exec_exception_subclass_occurred_no_message() {
     let err_code: i8 = 4;
-    let invalid_tx = create_throwing_exec_exception_mock_transaction_proxy(
+    let (invalid_tx, raw) = create_throwing_exec_exception_mock_transaction_proxy(
         EXECUTOR.clone(),
         true,
         err_code,
         None,
     );
     let db = MemoryDB::new();
     let mut fork = db.fork();
+    let tx_context = create_transaction_context(&mut fork, raw);
     let err = invalid_tx
-        .execute(&mut fork)
+        .execute(tx_context)
         .map_err(TransactionError::from)
         .expect_err("This transaction should be executed with an error!");
     assert_eq!(err.error_type(), TransactionErrorType::Code(err_code as u8));
     assert!(err.description().is_none());
 }
 
 #[test]
-fn json_serialize() {
-    let valid_tx = create_mock_transaction_proxy(EXECUTOR.clone(), true);
-    assert_eq!(valid_tx.serialize_field().unwrap(), *INFO_VALUE);
-}
-
-#[test]
-#[should_panic(expected = "Java exception: java.lang.OutOfMemoryError")]
-fn json_serialize_should_panic_if_java_error_occurred() {
-    let panic_tx = create_throwing_mock_transaction_proxy(EXECUTOR.clone(), OOM_ERROR_CLASS);
-    panic_tx.serialize_field().unwrap();
-}
-
-#[test]
-fn json_serialize_should_return_err_if_java_exception_occurred() {
-    let invalid_tx =
-        create_throwing_mock_transaction_proxy(EXECUTOR.clone(), ARITHMETIC_EXCEPTION_CLASS);
-    let err = invalid_tx
-        .serialize_field()
-        .expect_err("This transaction should be serialized with an error!");
-    assert!(err
-        .description()
-        .starts_with("Java exception: java.lang.ArithmeticException",));
+fn passing_transaction_context() {
+    let db = MemoryDB::new();
```

Comment:
The tests here are somewhat complex, I'd try to clarify what happens with comments, possibly, extracting some duplicate code.

Reply:
1. Such comments are not informative. Please comment _why_ this code does what it does. It is dead clear that we execute transaction, but that is done, if I understand correctly, to capture the context parameters and write them to the database. That is absolutely not clear from this code.

2. The other tests also need to be clarified.